### PR TITLE
INA228/INA229, INA238/INA239, INA237 power/energy/charge monitor (I2C, SPI) - Documentation

### PR DESCRIPTION
**Pull request by latonita**
## Description:
Documentation for component supporting INA228, INA229, INA238, INA239, INA237 power monitors.

**Related issue (if applicable):** fixes
https://github.com/esphome/feature-requests/issues/2549

**Pull request in [esphome](https://github.com/esphome/esphome) ** esphome/esphome#6138

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/ina2xx.rst [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+--------
+Use ``ina2xx_i2c`` component to connect INA228, INA237, or INA238 device over :ref:`I²C <i2c>` bus. 
+
+.. code-block:: yaml
+
+    # Example configuration entry for I²C
+    i2c:
+      sda: D1
+      scl: D2
+
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]: We do not include prerequisite component configuration in examples.
```suggestion
```

File: components/sensor/ina2xx.rst [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  - **shunt_voltage** (*Optional*, :ref:`config-time`): Conversion time for shunt voltage measurement.
+  - **temperature** (*Optional*, :ref:`config-time`): Conversion time for temperature measurement.
+
+- **adc_averaging** (*Optional*, integer): Selects ADC sample averaging count. Defaults to ``128``. 
+  Valid values are ``1``, ``4``, ``16``, ``64``, ``128``, ``256``, ``512``, ``1024``.
+- **temperature_coefficient** (*Optional*, integer from ``0`` to ``16383``): Temperature coefficient (ppm/°C) of the 
+  shunt for temperature compensation correction. Only applicable to INA228 and INA229 devices. Zero value means 
+  no compensation is done. Defaults to ``0``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- All other options for I²C device as descibed in respective documentation.
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]:
```suggestion
- All other options from :ref:`Sensor <config-sensor>` and :ref:`I²C device <i2c>`.
```

File: components/sensor/ina2xx.rst [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+
+.. code-block:: yaml
+
+    # Example configuration entry for SPI
+    spi:
+      clk_pin: D0
+      mosi_pin: D1
+      miso_pin: D2
+
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]: As above
```suggestion
```

File: components/sensor/ina2xx.rst [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    # Example configuration entry for SPI
+    spi:
+      clk_pin: D0
+      mosi_pin: D1
+      miso_pin: D2
+        
+    sensor:
+      - platform: ina2xx_spi
+        cs_pin: D3
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]: For pin numbers, we're now using this pattern (to force people to use their own pin(s)).
```suggestion
        cs_pin: GPIOXX
```

File: components/sensor/ina2xx.rst [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  - **shunt_voltage** (*Optional*, :ref:`config-time`): Conversion time for shunt voltage measurement.
+  - **temperature** (*Optional*, :ref:`config-time`): Conversion time for temperature measurement.
+
+- **adc_averaging** (*Optional*, integer): Selects ADC sample averaging count. Defaults to ``128``. 
+  Valid values are ``1``, ``4``, ``16``, ``64``, ``128``, ``256``, ``512``, ``1024``.
+- **temperature_coefficient** (*Optional*, integer from ``0`` to ``16383``): Temperature coefficient (ppm/°C) of the 
+  shunt for temperature compensation correction. Only applicable to INA228 and INA229 devices. Zero value means 
+  no compensation is done. Defaults to ``0``.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
+- All other options for SPI device as descibed in respective documentation.
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]:
```suggestion
- All other options from :ref:`Sensor <config-sensor>` and :ref:`SPI device <spi>`.
```

File: components/sensor/ina2xx.rst [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+
+Sensors
+-------
+The component offers nine sensors. You can configure all or any subset of the sensors. Each configured sensor 
+is reported  separately on each update_interval. The ``name`` option is required for each sensor configured. 
+All other options from :ref:`Sensor <config-sensor>`. Shorthand notation can be used for the sensors like this:
```
Comment (kbx81) [inline components/sensor/ina2xx.rst]:
```suggestion
is reported  separately on each update_interval. The ``name`` option is required for each sensor configured; 
all other options from :ref:`Sensor <config-sensor>` are also supported. Shorthand notation can be used for the sensors like this:
```

**Review by kbx81 — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**